topics/tutorials/custom_language_support/structure_aware_navbar.md

```diff
@@ -0,0 +1,37 @@
+[//]: # "title: 15. Structure Aware Navigation Bar"
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+Structure aware navbar allows displaying specific file elements in the [navigation bar](https://www.jetbrains.com/help/idea/guided-tour-around-the-user-interface.html#navigation-bar), depending on the location of the caret in it. For example, in Java this is used to display a class and method in which the caret locates.
+
+**Reference**: [Navigation Bar](navbar.md)
+
+## Define a SimpleStructureAwareNavbar
+
+The simple structure aware navbar implements [`StructureAwareNavBarModelExtension`](upsource:///platform/lang-impl/src/com/intellij/ide/navigationToolbar/StructureAwareNavBarModelExtension.java).
+
+Override the `getLanguage()` method to return the language for which the navigation bar will be used. To define text for a file and a property, override the `getPresentableText()` method, and finally, to display an icon for a property, override the `getIcon()` method.
+
+```java
+```
+{src="simple_language_plugin/src/main/java/org/intellij/sdk/language/SimpleStructureAwareNavbar.java"}
+
+In the `adjustElement()` method, if the current caret is in a comment, then we check if this comment is right above the property, we return this property. Thus, when the user selects a comment, the property name will be displayed in the navigation bar.
```

Comment:
if this comment is located just before a property and then return this property.